```
       last 25 log lines:
       > [ 33%] Building C object euicc/CMakeFiles/euicc.dir/interface.c.o
       > [ 35%] Building C object euicc/CMakeFiles/euicc.dir/sha256.c.o
       > [ 37%] Building C object euicc/CMakeFiles/euicc.dir/tostr.c.o
       > [ 40%] Linking C shared library libeuicc.dylib
       > ld: warning: ignoring file /nix/store/x25662zlrvzy1aavv8y5vrk40bj8c8hr-clang-wrapper-19.1.6/resource-root/lib/darwin/libclang_rt.osx.a, building for macOS-x86_64 but attempting to link with file built for Mac Catalyst-arm64
       > [ 40%] Built target euicc
       > [ 48%] Building C object driver/CMakeFiles/euicc-drivers.dir/apdu/pcsc.c.o
       > [ 48%] Building C object driver/CMakeFiles/euicc-drivers.dir/driver.c.o
       > [ 48%] Building C object driver/CMakeFiles/euicc-drivers.dir/http/curl.c.o
       > [ 48%] Building C object driver/CMakeFiles/euicc-drivers.dir/http/stdio.c.o
       > [ 51%] Building C object driver/CMakeFiles/euicc-drivers.dir/apdu/stdio.c.o
       > In file included from /tmp/nix-build-lpac-2.2.1.drv-0/source/driver/apdu/pcsc.c:12:
       > /nix/store/94mq060mc4p6knpwix5li0icpfsa07qq-pcsclite-2.3.0-dev/include/PCSC/winscard.h:41:10: error: 'pcsclite.h' file not found with <angled> include; use "quotes" instead
       >    41 | #include <pcsclite.h>
       >       |          ^
       > In file included from /tmp/nix-build-lpac-2.2.1.drv-0/source/driver/apdu/pcsc.c:12:
       > In file included from /nix/store/94mq060mc4p6knpwix5li0icpfsa07qq-pcsclite-2.3.0-dev/include/PCSC/winscard.h:41:
       > /nix/store/94mq060mc4p6knpwix5li0icpfsa07qq-pcsclite-2.3.0-dev/include/PCSC/pcsclite.h:45:10: error: 'wintypes.h' file not found with <angled> include; use "quotes" instead
       >    45 | #include <wintypes.h>
       >       |          ^
       > 2 errors generated.
       > make[2]: *** [driver/CMakeFiles/euicc-drivers.dir/build.make:121: driver/CMakeFiles/euicc-drivers.dir/apdu/pcsc.c.o] Error 1
       > make[2]: *** Waiting for unfinished jobs....
       > make[1]: *** [CMakeFiles/Makefile2:241: driver/CMakeFiles/euicc-drivers.dir/all] Error 2
       > make: *** [Makefile:136: all] Error 2
```

--- a/pkgs/by-name/lp/lpac/lpac-version.patch
+++ b/pkgs/by-name/lp/lpac/lpac-version.patch
@@ -1,0 +1,13 @@
+diff --git a/cmake/git-version.cmake b/cmake/git-version.cmake
+index be226fd..1451ff6 100644
+--- a/cmake/git-version.cmake
++++ b/cmake/git-version.cmake
+@@ -15,6 +15,8 @@ if(GIT_EXECUTABLE)
+   endif()
+ endif()
+ 
++set(LPAC_VERSION "$ENV{LPAC_VERSION}")
++
+ # Final fallback: Just use a bogus version string that is semantically older
+ # than anything else and spit out a warning to the developer.
+ if(NOT DEFINED LPAC_VERSION)

--- a/pkgs/by-name/lp/lpac/package.nix
+++ b/pkgs/by-name/lp/lpac/package.nix
@@ -1,0 +1,59 @@
+{
+  stdenv,
+  fetchFromGitHub,
+  lib,
+  cmake,
+  pkg-config,
+  pcsclite,
+  curl,
+  withDrivers ? true,
+  withLibeuicc ? true,
+  nix-update-script,
+}:
+
+let
+  inherit (lib) optional;
+in
+stdenv.mkDerivation (finalAttrs: {
+
+  pname = "lpac";
+  version = "2.2.1";
+
+  src = fetchFromGitHub {
+    owner = "estkme-group";
+    repo = "lpac";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-dxoYuX3dNj4piXQBqU4w1ICeyOGid35c+6ZITQiN6wA=";
+  };
+
+  env.LPAC_VERSION = finalAttrs.version;
+
+  patches = [ ./lpac-version.patch ];
+
+  cmakeFlags =
+    optional withDrivers "-DLPAC_DYNAMIC_DRIVERS=on"
+    ++ optional withLibeuicc "-DLPAC_DYNAMIC_LIBEUICC=on";
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+  ];
+
+  buildInputs = [
+    curl
+    pcsclite
+  ];
+
+  passthru = {
+    updateScript = nix-update-script { attrPath = finalAttrs.pname; };
+  };
+
+  meta = {
+    description = "C-based eUICC LPA";
+    homepage = "https://github.com/estkme-group/lpac";
+    mainProgram = "lpac";
+    license = [ lib.licenses.agpl3Plus ] ++ optional withLibeuicc lib.licenses.lgpl21Plus;
+    maintainers = with lib.maintainers; [ sarcasticadmin ];
+    platforms = lib.platforms.all;
+  };
+})

